tree/dataframe/inc/ROOT/RDF/RInterface.hxx

```diff
@@ -1335,29 +1335,75 @@ public:
-                                         colListWithAliasesAndSizeBranches, options});
-
       ::TDirectory::TContext ctxt;
 
-      // The CreateLMFromTTree function by default opens the file passed as input
-      // to check for the presence of the TTree inside. But at this moment the
-      // filename we are using here corresponds to a file which does not exist yet,
-      // i.e. the output file of the Snapshot call. Thus, checkFile=false will
-      // prevent the function from trying to open a non-existent file.
-      auto newRDF = std::make_shared<RInterface<RLoopManager>>(ROOT::Detail::RDF::CreateLMFromTTree(
-         fullTreeName, filename, colListNoAliasesWithSizeBranches, /*checkFile*/ false));
+      RResultPtr<RInterface<RLoopManager>> resPtr;
+
+      if (options.fOutputFormat == ESnapshotOutputFormat::kRNTuple) {
+#ifdef R__HAS_ROOT7
+         if (fLoopManager->GetTree()) {
+            throw std::runtime_error("Snapshotting from TTree to RNTuple is not yet supported. The current recommended "
+                                     "way to convert TTrees to RNTuple is through the RNTupleImporter.");
+         }
+
+         if (!dirname.empty()) {
+            throw std::runtime_error(
+               "RDataFrame: Snapshotting an RNTuple to a TFile sub-directory is currently not supported.");
+         }
+
+         // The data source of the RNTuple resulting from the Snapshot action does not exist yet here, so we create one
+         // without a data source for now, and set it once the actual data source can be created (i.e., after
+         // writing the RNTuple).
+         auto newRDF = std::make_shared<RInterface<RLoopManager>>(std::make_shared<RLoopManager>(colListNoPoundSizes));
+
+         auto snapHelperArgs = std::make_shared<RDFInternal::SnapshotHelperArgs>(RDFInternal::SnapshotHelperArgs{
+            std::string(filename), std::string(dirname), std::string(treename), colListWithAliasesAndSizeBranches,
+            options, newRDF->GetLoopManager(), true /* fToNTuple */});
+
+         // The Snapshot helper will use colListNoAliasesWithSizeBranches (with aliases resolved) as input columns, and
+         // colListWithAliasesAndSizeBranches (still with aliases in it, passed through snapHelperArgs) as output column
+         // names.
+         resPtr = CreateAction<RDFInternal::ActionTags::Snapshot, RDFDetail::RInferredType>(
+            colListNoAliasesWithSizeBranches, newRDF, snapHelperArgs, fProxiedPtr,
+            colListNoAliasesWithSizeBranches.size());
+#else
+         throw std::runtime_error(
+            "RDataFrame: Cannot snapshot to RNTuple - this installation of ROOT has not been build with ROOT7 "
+            "components enabled.");
+#endif
+      } else {
+         if (RDFInternal::GetDataSourceLabel(*this) == "RNTupleDS" &&
+             options.fOutputFormat == ESnapshotOutputFormat::kDefault) {
+            Warning("Snapshot",
```

Comment:
How does a user that is happy with the default (wants TTree) disable this warning?

Reply:
`fOutputFormat = kTTree` (modulo proper qualifying tokens)